core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1070,12 +1090,11 @@ object GroupMetadataManager {
    * @return payload for offset commit message
    */
   private[group] def offsetCommitValue(offsetAndMetadata: OffsetAndMetadata): Array[Byte] = {
-    // generate commit value with schema version 1
+    // generate commit value with schema version 2
     val value = new Struct(CURRENT_OFFSET_VALUE_SCHEMA)
-    value.set(OFFSET_VALUE_OFFSET_FIELD_V1, offsetAndMetadata.offset)
-    value.set(OFFSET_VALUE_METADATA_FIELD_V1, offsetAndMetadata.metadata)
-    value.set(OFFSET_VALUE_COMMIT_TIMESTAMP_FIELD_V1, offsetAndMetadata.commitTimestamp)
-    value.set(OFFSET_VALUE_EXPIRE_TIMESTAMP_FIELD_V1, offsetAndMetadata.expireTimestamp)
```

Comment:
Maybe I'm missing something, but don't we need to be able to use the old version of the offset commit schema if the expiration timestamp is provided by the user?